builtin/providers/google/resource_sql_database_instance_test.go

```diff
@@ -20,6 +21,105 @@ import (
 	"google.golang.org/api/sqladmin/v1beta4"
 )
 
+func init() {
+	resource.AddTestSweepers("gcp_sql_db_instance", &resource.Sweeper{
+		Name: "gcp_sql_db_instance",
+		F:    testSweepDatabases,
+	})
+}
+
+func testSweepDatabases(region string) error {
+	config, err := sharedConfigForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting shared config for region: %s", err)
+	}
+
+	err = config.loadAndValidate()
+	if err != nil {
+		log.Fatalf("error loading: %s", err)
+	}
+
+	found, err := config.clientSqlAdmin.Instances.List(config.Project).Do()
+	if err != nil {
+		log.Fatalf("error listing databases: %s", err)
+	}
+
+	if len(found.Items) == 0 {
+		log.Printf("No databases found")
+		return nil
+	}
+
+	for _, d := range found.Items {
+		var testDbInstance bool
+		for _, testName := range []string{"tf-lw-", "sqldatabasetest"} {
```

Comment:
Minor nit: This pattern happens a lot throughout all of the test sweepers. Perhaps extracting the logic to a function inside the `helper` package? 
```
func isTestResource(resourceName string, testNames []string) bool {
  // ...
}
```